pkg/services/publicdashboards/database/database_test.go

```diff
@@ -489,6 +488,42 @@ func TestIntegrationGetOrgIdByAccessToken(t *testing.T) {
 	})
 }
 
+func TestIntegrationDelete(t *testing.T) {
+	var sqlStore db.DB
+	var cfg *setting.Cfg
+	var dashboardStore *dashboardsDB.DashboardStore
+	var publicdashboardStore *PublicDashboardStoreImpl
+	var savedDashboard *models.Dashboard
+	var savedPublicDashboard *PublicDashboard
+
+	setup := func() {
+		sqlStore, cfg = db.InitTestDBwithCfg(t)
+		dashboardStore = dashboardsDB.ProvideDashboardStore(sqlStore, cfg, featuremgmt.WithFeatures(), tagimpl.ProvideService(sqlStore, cfg))
+		publicdashboardStore = ProvideStore(sqlStore)
+		savedDashboard = insertTestDashboard(t, dashboardStore, "testDashie", 1, 0, true)
+		savedPublicDashboard = insertPublicDashboard(t, publicdashboardStore, savedDashboard.Uid, savedDashboard.OrgId, true)
+	}
+
+	t.Run("Delete success", func(t *testing.T) {
+		setup()
+
+		err := publicdashboardStore.Delete(context.Background(), savedPublicDashboard.OrgId, savedPublicDashboard.DashboardUid, savedPublicDashboard.Uid)
+		require.NoError(t, err)
+
+		deletedDashboard, err := publicdashboardStore.FindByDashboardUid(context.Background(), savedPublicDashboard.OrgId, savedPublicDashboard.DashboardUid)
+		require.Error(t, err)
+		require.Equal(t, ErrPublicDashboardNotFound, err)
+		require.Nil(t, deletedDashboard)
+	})
+
+	t.Run("Non-existent public dashboard deletion doesn't throw an error", func(t *testing.T) {
```

Comment:
Maybe this should throw an error if you try to delete a dashboard that doesn't exist?

Reply:
we already talked about deletion process with @evictorero . there is no need right now and I followed the same behavior I found in Grafana project, where deletion doesn't throw an error if there was no deleted register.

Reply:
We talked about this for find not delete. We should probably at least return a boolean so that the service/caller can determine if a noop is an error

Reply:
ok gonna make it in another pr in order to merge it